mono/metadata/threads.c

```diff
@@ -2554,12 +2579,10 @@ mono_thread_suspend (MonoInternalThread *thread)
 }
 
 void
-ves_icall_System_Threading_Thread_Suspend (MonoThread *this_obj)
+ves_icall_System_Threading_Thread_Suspend (MonoThreadObjectHandle this_obj, MonoError *error)
 {
-	if (!mono_thread_suspend (this_obj->internal_thread)) {
-		mono_set_pending_exception (mono_get_exception_thread_state ("Thread has not been started, or is dead."));
-		return;
-	}
+	if (!mono_thread_suspend (thread_handle_to_internal_ptr (this_obj)))
+		mono_set_pending_exception_handle (mono_exception_new_thread_state ("Thread has not been started, or is dead.", error));
```

Comment:
Same as above - just set the MonoError via `mono_error_set_generic`

Reply:
I was trying to stay close to original and didn't notice that option. I'll look.
extern is pretty much meaningless noise in C and C++ and should be generally omitted as such. Like "auto" (historically, not lately).